overlay/overrides.nix

```
@@ -122,21 +127,18 @@ in rec {
     };
   };
 
-  pkg-config = if pkgs.stdenv.hostPlatform != pkgs.stdenv.buildPlatform
-    then makeOverride {
-      name = "pkg-config";
-      overrideAttrs = drv: {
-        propagatedBuildInputs = drv.propagatedBuildInputs or [ ] ++ [
-          (propagateEnv "pkg-config" [
-            { name = "PKG_CONFIG_ALLOW_CROSS"; value = "1"; }
-          ])
-        ];
-        propagatedNativeBuildInputs = drv.propagatedNativeBuildInputs or [ ] ++ [
-          pkgs.buildPackages.pkg-config
-        ];
-      };
-    }
-    else nullOverride;
+  pkg-config = makeOverride {
+    name = "pkg-config";
+    overrideAttrs = drv: {
+      propagatedBuildInputs = drv.propagatedBuildInputs or [ ] ++ [
+        pkgs.pkg-config
+      ] ++ lib.optional (pkgs.stdenv.hostPlatform != pkgs.stdenv.buildPlatform) [
+        (propagateEnv "pkg-config" [
+          { name = "PKG_CONFIG_ALLOW_CROSS"; value = "1"; }
+        ])
+      ];
```

We don't need to pass a list to `optional`,  as it already returns a list.
`optional = cond: elem: if cond then [elem] else [];`
```suggestion
      ] ++ lib.optional (pkgs.stdenv.hostPlatform != pkgs.stdenv.buildPlatform)
        (propagateEnv "pkg-config" [
          { name = "PKG_CONFIG_ALLOW_CROSS"; value = "1"; }
        ])
      ;
```

---

Good catch, man! I didn't realize this.